native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2TargetHandler.java

```diff
@@ -265,8 +265,11 @@ private void onResetRead(Http2Reset http2Reset) {
         int streamId = http2Reset.getStreamId();
         OutboundMsgHolder outboundMsgHolder = http2ClientChannel.getInFlightMessage(streamId);
         if (outboundMsgHolder != null) {
-            outboundMsgHolder.getResponseFuture()
-                    .notifyHttpListener(new Exception("HTTP/2 stream " + streamId + " reset by the remote peer"));
+            Http2MessageStateContext messageStateContext =
+                    outboundMsgHolder.getRequest().getHttp2MessageStateContext();
```

Comment:
For RSTStream, GOAWAY scenarios, the error will be the same as we call `handleConnectionClose` in every scenario. If we need to show custom errors, we may have to add more state events
WDYT @TharmiganK ?

Reply:
It is crucial to identify the reason for the connection closure when we debug issues. So +1 to add custom errors or add a reason for the error message. For example:
```
Remote host closed the connection with GOAWAY before initiating inbound response
Remote host closed the connection with RST before reading inbound response headers
```

Reply:
Updated